presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/MetastoreUtil.java

```diff
@@ -478,6 +479,12 @@ public static void verifyOnline(SchemaTableName tableName, Optional<String> part
             throw new TableOfflineException(tableName, false, null);
         }
 
+        // Throw table offline exception for un-partitioned tables when table parameters has object_not_readable
+        String objectNotReadable = parameters.get(OBJECT_NOT_READABLE);
+        if (!isNullOrEmpty(objectNotReadable) && !isPartitioned) {
```

Comment:
There are two scenarios 
- Table has not readable flag and only of 1 of the N partitions of the table are not readable
- Table has not readable flag and ALL of the partitions are also readable.

So for a partitioned table, I am not sure what object notreadble flag mean in table parameters.
FOr un-partitioned table, its pretty clear that the table is not readable..because no partitions exist.